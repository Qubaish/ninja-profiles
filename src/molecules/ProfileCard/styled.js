import styled, { keyframes } from 'styled-components';
import { device } from '../../utils/breakpoints';
import { flipInX } from "react-animations";

const FlipInXAnimation = keyframes`${flipInX}`;

export const StyledProfileCard = styled.div`
    width: 80%;
    padding: 0.5rem;
    background: #fff;
    animation: ${props => props.sec}s ${FlipInXAnimation};
    transition: all .3s ease;
    border: 1px solid #dcdde0;
    box-shadow: 1px 2px 3px rgba(220,221,224,0.6);
    border-radius: 0.3rem;
    box-sizing: border-box;

    :hover {
        transform: scale(1.06);
    }
    @media ${device.sm} { 
        width: 45%;
    }
    @media ${device.lg} { 
        width: 23%;
    }
`;

export const StyledContactInfo = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    padding-top: 1rem;
    a {
        color: #000;
        padding-right: 0.5rem;
    }
    h2 {
        margin: 0;
        font-size: 1.4rem;
    }
    p {
        font-size: 1.4rem;
        margin-top: 0.3rem;
    }
`;

export const StyledPersonalDetails = styled.div``;