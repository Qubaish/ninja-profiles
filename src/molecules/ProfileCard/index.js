import React from 'react';
import PropTypes from 'prop-types';
import { faLinkedin, faGithub, faTwitter } from '@fortawesome/free-brands-svg-icons';
import LazyImage from '../../atoms/LazyImage';
import Icons from '../../atoms/Icons';
import { StyledProfileCard, StyledPersonalDetails, StyledContactInfo } from './styled';
import { lqip, officeText } from '../../utils/constant';

const iconsToShow = {
    linkedIn: faLinkedin,
    gitHub: faGithub,
    twitter: faTwitter
};

const iconsToShowf = (props) => {
    const data = Object.keys(props).map(icon => {
        return props[icon] ? {
            icon: iconsToShow[icon],
            href: props[icon],
            name: icon,
        } : null;
    }).filter(x => x);
    return data;
};

export default function ProfileCard({ className, profile}) {
    const { name, office, imagePortraitUrl, linkedIn, twitter, gitHub } = profile;

    return (
        <StyledProfileCard className={className} sec={3}>
            <LazyImage
                className="avatar"
                aspectRatio={1 / 1}
                lqip={lqip}
                src={imagePortraitUrl}
                alt={`${name} ninja profile`}
            />
            <StyledContactInfo>
                <StyledPersonalDetails>
                    <h2>{name}</h2>
                    <p>{`${officeText} ${office}`}</p>
                </StyledPersonalDetails>
                <Icons icons={iconsToShowf({linkedIn, gitHub, twitter})} />
            </StyledContactInfo>
        </StyledProfileCard>
    );
}

ProfileCard.propTypes = {
    className: PropTypes.string,
    profile: PropTypes.shape({}),
};