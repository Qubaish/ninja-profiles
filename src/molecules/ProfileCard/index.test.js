import React from 'react';
import { render } from '@testing-library/react';
import ProfileCard from './index';

const mockProfile = {
    "name": "Simon",
    "office": "LUND",
    "gitHub": "simonpal",
    "twitter": "simonpal",
    "linkedIn": "/in/test/",
    "imagePortraitUrl": "img",
}

describe('<ProfileCard />', () => {
    it('checking snapshot', () => {
        const { asFragment } = render(<ProfileCard className="card" profile={mockProfile}/>);
        expect(asFragment()).toMatchSnapshot();
    });
    it('renders profile office text', () => {
        const { getByText } = render(<ProfileCard className="card" profile={mockProfile}/>);
        const linkElement = getByText(/LUND/i);
        expect(linkElement).toBeInTheDocument();
    });
    it('renders profile name', () => {
        const { getByText } = render(<ProfileCard className="card" profile={mockProfile}/>);
        const elem = getByText(/Simon/i);
        expect(elem).toBeInTheDocument();
    });
});

