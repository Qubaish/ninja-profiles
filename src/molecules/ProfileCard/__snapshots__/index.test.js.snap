// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileCard /> checking snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-AxiKw bcwudH card"
  >
    <div
      class="sc-AxjAm hfotVk avatar"
    >
      <div
        data-testid="ratio"
        style="padding-bottom: 100%;"
      />
      <img
        alt=""
        aria-hidden="true"
        class="sc-AxirZ lgwoU"
        src="data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAABYAAAAeCAIAAACnhXkBAAAACXBIWXMAAC4jAAAuIwF4pT92AAAEtElEQVQ4y32UeyyXbRjHn9F6nZscRk7lfMqyDorlD23NNGxmMWMjciiHUFoIm0NWTTYbbVmz5hhhjHJIOQwVWns1h8SIEIkSK/K+n/f3vFOvft77j3v389z393tf1/f6Xrfwl7SxsbHx48cPZqmfW4awHV5cr6ysfPv2bfO/VBbhdzzz2tpaR0dHbm5uTk7O3bt3a2pqBgYG+Ll5YFsKcXtycvLmzZvh4eE3btyoqqpqamqqr6+/f/9+bW3thw8ffmcRtuBHRkaio6NjYmLu3LlTV1fX09Pzp2S8fPmyubm5rKzs06dPW1iEX/Ffv35NSUm5dOlSdnZ2dXX1s2fPXr16NTQ0NDExMTY2Njw8/FQypFOI+nV3d1+8ePHatWv5+fmPHz/u6+t7/fr19PT08vIyl4+Ojvb39z969Ojz58+/svyHAtmIAgnu3bvX2Nj44sWLwcHBjx8/IuTq6ioavXnzpqGhYXZ2dluKysrKkJAQT0/PCxculJeX9/b2IgG5PHnypLW19e3btwSCQHBJoRC/S0tL/f39nZycrKysAgMD0Y87k5KSrl+/Xlxc3N7eTmkfPHjw/v17KRTiYBstYHFxcbGzs0NRTlNgQnv48OG7d++QFqURaFsKoqCWpJCXl+fn54cpZmZmOjs7KyoqEGV+fh4tioqKyGgrxWZF8WJhYSGZZ2ZmXrlyhaLipfHxcWKZmppaWlqirmgxNzcnnQLlyRlTp6amnj592svL69atWwiMqFQHCuqK03DqwsLCthSXL19OTEx0dnY+evToqVOnQkNDz549S15oScvgmpKSEmz+/fv3rQYXKag8GF9fXyqCnOfOneMTdcPCwlgjB2bhAvwmxZ2bXXz16tWTJ0+eOHECMOlkZWVRCOoSEBAQFxfHblRU1BZf/azI+vo6M6Y+cOAAKdBp6enpaWlplBlTFxQUeHt7u7m5BQcHo/r/tRmDUI8fP37mzJmEhATMzmNx+/ZtqoNl7e3tDx8+/Pz5c/FB4VYR9W8i/BJfJ2RzcHBwd3c/f/48LUsssbGxPj4+RBcUFESA2ATVRDAzRALgzW+CRDyicHV1PXLkCKWhtI6Ojvv370dm5MQvVHRxcbGrqwu//IyCXm5paUEzUuU2KJKTk8FbWlqamJhAR+/iN94hZlELZDI2NqbkyCRkZGTQDjt37lRSUsIX2trae/fuJX5eDSInC16qtrY2NOICDIrNaWt2dXV1zc3NIRJkZWXl5OTk5eWtra35/kMyAHMzilJXikLjHDx4EF9++fIFs/Oss2VgYECyCCzs2rVLWVkZmL6+Pmu4NTQ0iCs+Pp4AKQdRYDYPDw+6CzAsJEKYenp6tra2hw4dEghBQUFBnCEiHR0dHTMzMzB4iYcbLhsbG3Q5duxYZGQkKdA4ERERUPAfpQUxC5GFBSzMhIOuJI9HOaSmpiYjIyNIxo4dO1RVVUnZXDLQWxAxioqKKioq7Kmrq2tpaWlqanKIe8AQF9mZmpoCsLCw4EFDNRFsZGT0zxkuRwvAIA0NDUmBiohEZMSaFAAwgwQvEsG4b9++PXv2UEEBPPpxGj4AsCC1nmRwCEau4kKQwFiYSQbHOLB7924o/gZ2se834PBMuwAAAABJRU5ErkJggg=="
      />
      <img
        alt="Simon ninja profile"
        class="sc-AxirZ lgwoU source"
        loading="lazy"
        src="img"
      />
    </div>
    <div
      class="sc-AxhCb eQqVtK"
    >
      <div
        class="sc-AxhUy kMyssT"
      >
        <h2>
          Simon
        </h2>
        <p>
          Office: LUND
        </p>
      </div>
      <div>
        <a
          alt="linkedIn"
          data-testid="icon-0"
          href="/in/test/"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-linkedin fa-w-14 "
            data-icon="linkedin"
            data-prefix="fab"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
              fill="currentColor"
            />
          </svg>
        </a>
        <a
          alt="gitHub"
          data-testid="icon-1"
          href="simonpal"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-github fa-w-16 "
            data-icon="github"
            data-prefix="fab"
            focusable="false"
            role="img"
            viewBox="0 0 496 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M165.9 397.4c0 2-2.3 3.6-5.2 3.6-3.3.3-5.6-1.3-5.6-3.6 0-2 2.3-3.6 5.2-3.6 3-.3 5.6 1.3 5.6 3.6zm-31.1-4.5c-.7 2 1.3 4.3 4.3 4.9 2.6 1 5.6 0 6.2-2s-1.3-4.3-4.3-5.2c-2.6-.7-5.5.3-6.2 2.3zm44.2-1.7c-2.9.7-4.9 2.6-4.6 4.9.3 2 2.9 3.3 5.9 2.6 2.9-.7 4.9-2.6 4.6-4.6-.3-1.9-3-3.2-5.9-2.9zM244.8 8C106.1 8 0 113.3 0 252c0 110.9 69.8 205.8 169.5 239.2 12.8 2.3 17.3-5.6 17.3-12.1 0-6.2-.3-40.4-.3-61.4 0 0-70 15-84.7-29.8 0 0-11.4-29.1-27.8-36.6 0 0-22.9-15.7 1.6-15.4 0 0 24.9 2 38.6 25.8 21.9 38.6 58.6 27.5 72.9 20.9 2.3-16 8.8-27.1 16-33.7-55.9-6.2-112.3-14.3-112.3-110.5 0-27.5 7.6-41.3 23.6-58.9-2.6-6.5-11.1-33.3 2.6-67.9 20.9-6.5 69 27 69 27 20-5.6 41.5-8.5 62.8-8.5s42.8 2.9 62.8 8.5c0 0 48.1-33.6 69-27 13.7 34.7 5.2 61.4 2.6 67.9 16 17.7 25.8 31.5 25.8 58.9 0 96.5-58.9 104.2-114.8 110.5 9.2 7.9 17 22.9 17 46.4 0 33.7-.3 75.4-.3 83.6 0 6.5 4.6 14.4 17.3 12.1C428.2 457.8 496 362.9 496 252 496 113.3 383.5 8 244.8 8zM97.2 352.9c-1.3 1-1 3.3.7 5.2 1.6 1.6 3.9 2.3 5.2 1 1.3-1 1-3.3-.7-5.2-1.6-1.6-3.9-2.3-5.2-1zm-10.8-8.1c-.7 1.3.3 2.9 2.3 3.9 1.6 1 3.6.7 4.3-.7.7-1.3-.3-2.9-2.3-3.9-2-.6-3.6-.3-4.3.7zm32.4 35.6c-1.6 1.3-1 4.3 1.3 6.2 2.3 2.3 5.2 2.6 6.5 1 1.3-1.3.7-4.3-1.3-6.2-2.2-2.3-5.2-2.6-6.5-1zm-11.4-14.7c-1.6 1-1.6 3.6 0 5.9 1.6 2.3 4.3 3.3 5.6 2.3 1.6-1.3 1.6-3.9 0-6.2-1.4-2.3-4-3.3-5.6-2z"
              fill="currentColor"
            />
          </svg>
        </a>
        <a
          alt="twitter"
          data-testid="icon-2"
          href="simonpal"
          target="_blank"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-twitter fa-w-16 "
            data-icon="twitter"
            data-prefix="fab"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M459.37 151.716c.325 4.548.325 9.097.325 13.645 0 138.72-105.583 298.558-298.558 298.558-59.452 0-114.68-17.219-161.137-47.106 8.447.974 16.568 1.299 25.34 1.299 49.055 0 94.213-16.568 130.274-44.832-46.132-.975-84.792-31.188-98.112-72.772 6.498.974 12.995 1.624 19.818 1.624 9.421 0 18.843-1.3 27.614-3.573-48.081-9.747-84.143-51.98-84.143-102.985v-1.299c13.969 7.797 30.214 12.67 47.431 13.319-28.264-18.843-46.781-51.005-46.781-87.391 0-19.492 5.197-37.36 14.294-52.954 51.655 63.675 129.3 105.258 216.365 109.807-1.624-7.797-2.599-15.918-2.599-24.04 0-57.828 46.782-104.934 104.934-104.934 30.213 0 57.502 12.67 76.67 33.137 23.715-4.548 46.456-13.32 66.599-25.34-7.798 24.366-24.366 44.833-46.132 57.827 21.117-2.273 41.584-8.122 60.426-16.243-14.292 20.791-32.161 39.308-52.628 54.253z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
